monitor/monitor.go

```diff
@@ -54,28 +51,32 @@ func New(cfg *build.InvokeConfig, stdin io.ReadCloser, stdout, stderr io.WriteCl
 
 func (m *Monitor) Handler() build.Handler {
 	return build.Handler{
-		OnResult: func(driverIndex int, gotRes *build.ResultHandle) {
-			m.mu.Lock()
-			defer m.mu.Unlock()
+		Evaluate: func(ctx context.Context, c gateway.Client, res *gateway.Result) error {
```

Comment:
As this doesn't use any local variables in the closure, move this to actual private method. Or you can just add `Evaluate` to `Monitor` and pass it as handler.